qlib/kernel/asm/aarch64.rs

```diff
@@ -107,19 +107,30 @@ pub fn IRet(kernelRsp: u64) -> ! {
 }
 
 #[inline]
-pub fn GetRsp() -> u64 {
-    let rsp: u64;
+pub fn GetCurrentKernelSp() -> u64 {
+    // we can only mrs sp_el1 in EL2 and EL3
+    // so we can only get sp_el1 by move
+    let ret: u64;
     unsafe {
-        asm!("mov {0}, sp", out(reg) rsp);
-    };
-    return rsp;
+        asm!("\
+        mrs x20, spsel
```

Comment:
If I understand correctly x20 here is used to save the original value of spsel.
However I think this may breaks the calling convention because x20 is callee saved.

To explicitly tell the compiler to "clobber" the register, maybe add some thing like:
```rust
out("x20") _,
```

Reply:
yes, you are right, and modified to use a scratch register